### ISSUE TITLE
scss one line comments issue with rule-nested-empty-line-before : always

### ISSUE THREAD

**Issue by mynameislau** [stylelint/stylelint#845]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Bug where a line of scss comments (//) triggers the expected rule-nested-empty-line-before rule, when it shouldn't.

> Which rule, if any, is this issue related to?

rule-nested-empty-line-before

> What CSS is needed to reproduce this issue?

.content {
  //min-height: 100%;

  &.content--neutral {
    background-color: $grey;
  }
}

> What stylelint configuration is needed to reproduce this issue?

e.g.

rule-nested-empty-line-before: "always"

> Which version of stylelint are you using?

4.4.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Node

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes it is related to SCSS

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

Expected empty line before nested rule (rule-nested-empty-line-before)

---

Note: stackoverflow is our preferred QA forum - http://stackoverflow.com/questions/tagged/stylelint

**Comment by jeddy3** [comment]
@mynameislau Thanks for the detailed report.

I'm unable to reproduce the problem though.

This SCSS:

```scss
.content {
//min-height: 100%;

&.content--neutral {
background-color: $grey;
}
}
```

With this this config:

```json
{ 
  "rules": {
    "rule-nested-empty-line-before": "always"
  }
}
```

And with this node API usage:

```js
var stylelint = require('stylelint')
stylelint.lint({ files: "css/scss.css", syntax: "scss" })
  .then(function(data) { console.log(data.output) })
  .catch(function(err) { console.error(err.stack) })
```

Doesn't produce any warnings for me.

Can you double check that this is warning is being flagged against that particular CSS?

**Comment by jeddy3** [comment]
I'm going to close this issue for now. @mynameislau Feel free to reopen it if you're able to create a reduced test case for us.

**Comment by mynameislau** [comment]
Ok sorry for the delay, i'll look further into it as soon as I can.

**Comment by mynameislau** [comment]
[reduced.zip](https://github.com/stylelint/stylelint/files/158698/reduced.zip)

Here is a reduced test case, you should see 5:3 Expected empty line before nested rule (rule-nested-empty-line-before)

**Comment by jeddy3** [comment]
@mynameislau Thanks for the reduced test case.

This was a weird one to track down... 

I'm guessing you're on working on a windows machine as you're using CRLF line ends? It looks like `postcss-scss` parser is changing the line ending after the comment. It introduces a mismatch of line endings (`\n` after the comment, follow by a `\r\n` for the empty line), which means the rule doesn't register it as a empty line.

@davidtheclark Should we raise this upstream or should we patch it at our end by simple adding `\n\r\n` to our list of empty lines?

**Comment by davidtheclark** [comment]
> @davidtheclark Should we raise this upstream or should we patch it at our end by simple adding \n\r\n to our list of empty lines?

Probably we should do both. Patch it on our end and file an issue to see if postcss-scss wants to address this.

**Comment by ai** [comment]
PostCSS SCSS 0.1.7 fixes this issue.

**Comment by davidtheclark** [comment]
Thanks! @mynameislau can you please re-install your `node_modules` and see if the problem goes away?

**Comment by mynameislau** [comment]
@davidtheclark I confirm, the problem goes away :) :+1:
